src/renderers/shared/fiber/ReactFiberCompleteWork.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ReactFiberCompleteWork
+ * @flow
+ */
+
+'use strict';
+
+import type { ReactCoroutine, ReactYield } from 'ReactCoroutine';
+import type { Fiber } from 'ReactFiber';
+
+var ReactChildFiber = require('ReactChildFiber');
+var ReactTypesOfWork = require('ReactTypesOfWork');
+var {
+  IndeterminateComponent,
+  FunctionalComponent,
+  ClassComponent,
+  HostComponent,
+  CoroutineComponent,
+  YieldComponent,
+} = ReactTypesOfWork;
+
+function transferOutput(unitOfWork : Fiber) {
+  var child = unitOfWork.child;
+  if (child && !child.sibling) {
+    // Common case
+    unitOfWork.output = child.output;
+    return;
+  }
+  var output = [];
+  while (child) {
+    output.push(child.output);
+    child = child.sibling;
+  }
+  unitOfWork.output = output;
+}
+
+function recursivelyFillYields(yields, output) {
+  if (Array.isArray(output)) {
+    for (var i = 0; i < output.length; i++) {
+      recursivelyFillYields(yields, output[i]);
+    }
+  } else {
+    yields.push(output);
+  }
+}
+
+function handleCoroutine(unitOfWork : Fiber) {
+  var coroutine = (unitOfWork.input : ?ReactCoroutine);
+  if (!coroutine) {
+    throw new Error('Should be resolved by now');
+  }
+
+  // Hacky McHack
+  if (!unitOfWork.hasPendingChanges) {
+    return null;
+  }
+  unitOfWork.hasPendingChanges = false;
+
+  var yields = [];
+  var child = unitOfWork.child;
+  while (child) {
+    recursivelyFillYields(yields, child.output);
+    child = child.sibling;
+  }
+  var fn = coroutine.handler;
+  var props = coroutine.props;
+  var nextChildren = fn(props, yields);
```

Comment:
I'm not exactly sure what you mean so I'll throw out some words.

The yields have their fiber resolved at the location they're yielded. Meaning that their state identity is determined by where it is created, not where it ends up.

The memoization of the continuation will be stored on that fiber. If you render the continuation in multiple places, this breaks down but I don't plan on supporting that initially. It'd need to be cloned or immutable.